tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -2590,39 +2719,46 @@ class D3D12Device : public RendererBase
             // table).
             //
             auto& rootParamIndex = ioOffset.rootParam;
-            DescriptorSet descriptorSet;
 
-            if(auto descriptorCount = specializedLayout->getTotalResourceDescriptorCount())
+            if (auto descriptorCount = specializedLayout->getTotalResourceDescriptorCount())
             {
                 // There is a non-zero number of resource descriptors needed,
                 // so we will allocate a table out of the appropriate heap,
                 // and store it into the appropriate part of `descriptorSet`.
                 //
-                auto descriptorHeap = &transientHeap->m_viewHeap;
-                auto& table = descriptorSet.resourceTable;
+                auto descriptorHeap = &transientHeap->getCurrentViewHeap();
+                auto& table = outDescriptorSet.resourceTable;
 
                 // Allocate the table.
                 //
-                table.allocate(descriptorHeap, descriptorCount);
+                if (!table.allocate(descriptorHeap, descriptorCount))
+                {
+                    context->outOfMemoryHeap = D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV;
```

Comment:
Should handling the out-of-memory case be callback-oriented in the future? I'm not clear if it is better to let the application intercept this and try to allow the app to solider on, or whether it is always easier to have the operation fail and then just retry at the application level.